pkg/kubelet/kubelet.go

```diff
@@ -893,8 +894,17 @@ func (kl *Kubelet) getServiceEnvVarMap(ns string) (map[string]string, error) {
 	return m, nil
 }
 
-// Make the service environment variables for a pod in the given namespace.
-func (kl *Kubelet) makeEnvironmentVariables(ns string, container *api.Container) ([]string, error) {
+func (kl *Kubelet) getDownwardAPIEnvVarMap(pod *api.Pod) map[string]string {
+	return map[string]string{
+		"K8S_API_POD_NAME": pod.Name,
```

Comment:
If what is needed is just a unique ID, how about we add PodHostname to PodSpec, default it to "pod" suffixed by a hash of the namespace and uid, and use that to set the hostname inside the pod, which can then be used as a unique ID. If the one wanted to replace the pod but carry over the same PodHostname, that would at least be possible.

What other values need to be passed via the downward API in the short term?